tests/SprykerEcoTest/Zed/Adyen/Business/Facade/AdyenFacadeExecutePostSaveHookTest.php

```diff
@@ -0,0 +1,256 @@
+    }
+
+    /**
+     * @dataProvider refusalStatusDataProvider
+     *
+     * @param string $refusalStatus
+     *
+     * @return void
+     */
+    public function testExecutePostSaveHookShouldProcessErrorsWhenResponsIsSuccessAndHasRefusalStatus(string $refusalStatus): void
+    {
+        //Arrange
+        $adyenApiResponseTransfer = (new AdyenApiResponseBuilder([
+                AdyenApiResponseTransfer::IS_SUCCESS => true,
+            ]))
+            ->withMakePaymentResponse([
+                AdyenApiMakePaymentResponseTransfer::RESULT_CODE => $refusalStatus,
+                AdyenApiMakePaymentResponseTransfer::DETAILS => static::MAKE_PAYMENT_RESPONSE_DETAILS,
+                AdyenApiMakePaymentResponseTransfer::PAYMENT_DATA => static::MAKE_PAYMENT_RESPONSE_PAYMENT_DATA,
+            ])
+            ->withError()
+            ->build();
+
+        $adyenToAdyenApiFacadeBridgeMock = $this->createMock(AdyenToAdyenApiFacadeBridge::class);
+        $adyenToAdyenApiFacadeBridgeMock->method('performMakePaymentApiCall')
+            ->willReturn($adyenApiResponseTransfer);
+
+        $facade = $this->createFacade($adyenToAdyenApiFacadeBridgeMock);
+
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW,
+        );
+
+        $checkoutResponseTransfer = $this->tester->createCheckoutResponseTransfer();
+
+        //Act
+        $facade->executePostSaveHook(
+            $this->tester->buildQuoteTransfer($this->getPaymentAdyenReference($orderTransfer->getIdSalesOrder())),
+            $checkoutResponseTransfer,
+        );
+
+        //Assert
+        $this->assertCount(1, $checkoutResponseTransfer->getErrors());
+
+        $this->assertFalse($checkoutResponseTransfer->getIsSuccess());
+
+        $checkoutErrorTransfer = $checkoutResponseTransfer->getErrors()->getIterator()->current();
+        /** @var \Generated\Shared\Transfer\CheckoutErrorTransfer $checkoutErrorTransfer */
+        $this->assertSame(static::ERROR_CODE_PAYMENT_FAILED, $checkoutErrorTransfer->getErrorCode());
+        $this->assertSame(static::ERROR_TYPE_PAYMENT_FAILED, $checkoutErrorTransfer->getErrorType());
+
+        $spyPaymentAdyenApiLog = $this->getLastSpyPaymentAdyenApiLog();
+        $this->assertNotNull($spyPaymentAdyenApiLog);
+        $this->assertSame(static::MAKE_PAYMENT_PAY_PAL_REQUEST_TYPE, $spyPaymentAdyenApiLog->getType());
+        $this->assertTrue($spyPaymentAdyenApiLog->getIsSuccess());
+
+        $spyPaymentAdyen = $this->getSpyPaymentAdyen($orderTransfer);
+        $this->assertNotNull($spyPaymentAdyen);
+        $this->assertSame(static::MAKE_PAYMENT_RESPONSE_PAYMENT_DATA, $spyPaymentAdyen->getPaymentData());
+        $this->assertSame(json_encode(static::MAKE_PAYMENT_RESPONSE_DETAILS), $spyPaymentAdyen->getDetails());
+    }
+
+    /**
+     * @return array<array>
+     */
+    public function refusalStatusDataProvider(): array
```

Comment:
please add also test(s) covering exception throwing for required transfer properties

Reply:
The tests have been added as suggested (the test class was renamed):
```
ExecutePostSaveHookTest::testExecutePostSaveHookWithoutMakePaymentResponse
ExecutePostSaveHookTest::testExecutePostSaveHookWithoutPayment
ExecutePostSaveHookTest::testExecutePostSaveHookWithoutAdyenPaymen
ExecutePostSaveHookTest::testExecutePostSaveHookWithoutPaymentSelection
ExecutePostSaveHookTest::testExecutePostSaveHookWithoutBillingAddress
```